### ISSUE TITLE
[Image] = operator doesn't seem to copy the image

### ISSUE THREAD

**Issue by archanasri** [SCIInstitute/ShapeWorks#944]
<img width="1079" alt="Screen Shot 2021-02-05 at 9 01 15 PM" src="https://user-images.githubusercontent.com/31980477/107071189-668b8480-67f5-11eb-86d9-759b310daf86.png">

In the screenshot, img2.recenter() changes the origin for both img1 and img2. It should change the origin for only img2.

**Comment by archanasri** [comment]
Looks like an issue with the python binding. The same test passes from the C++ API.

**Comment by medakk** [comment]
FWIW, I think this is the expected behavior in Python. `A=B` just makes both of them refer to the same object, whether its a list, dictionary, etc. Even in numpy:

```python
[1]: import numpy as np

In [2]: A = np.eye(3)

In [3]: A
Out[3]: 
array([[1., 0., 0.],
       [0., 1., 0.],
       [0., 0., 1.]])

In [4]: B = A

In [5]: B *= 5

In [6]: B
Out[6]: 
array([[5., 0., 0.],
       [0., 5., 0.],
       [0., 0., 5.]])

In [7]: A
Out[7]: 
array([[5., 0., 0.],
       [0., 5., 0.],
       [0., 0., 5.]])

```

Copies are explicit, like `B = A.copy()` in numpy.

(Something like `C = 5*A` will construct a new object however.)

**Comment by cchriste** [comment]
> FWIW, I think this is the expected behavior in Python. `A=B` just makes both of them refer to the same object, whether its a list, dictionary, etc. Even in numpy:

Exactly. Python aims to be smart and know when something can simply be referenced, but makes a copy when either of the original objects changes. This either 1) isn't always desired, as in the numpy example Karthik shared, or 2) isn't easy to detect, as in the example Archana provided for this issue. 

There's a module called `copy`, and I tried it, but it expects objects to be able to be pickled. So we'll just make copying explicit, which emulates numpy's strategy.

**Comment by cchriste** [comment]
I've added `Image.copy` and `Mesh.copy` (along with the missing Mesh copy/assignment functions) to the Python api along with tests for both. See mesh_party for their implementation.

resolved in b3185ede

**Comment by archanasri** [comment]
Fix merged to master.
